src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,445 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Contains checks that are not architecture-specific
+
+#![allow(dead_code)]
+
+use crate::args::EnvArgument;
+use crate::ops::version;
+//use anyhow::Result;
+
+use crate::arch::arch_specific;
+use crate::check;
+use crate::utils;
+use kata_types::config::TomlConfig;
+
+use anyhow::{anyhow, Context, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use sys_info;
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct HostInfo {
+    #[serde(default)]
+    available_guest_protection: String,
+    #[serde(default)]
+    kernel: String,
+    #[serde(default)]
+    architecture: String,
+    #[serde(default)]
+    vm_container_capable: bool,
+    #[serde(default)]
+    support_v_socks: bool,
+    #[serde(default)]
+    distro: DistroInfo,
+    #[serde(default)]
+    cpu: CPUInfo,
+    #[serde(default)]
+    memory: MemoryInfo,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct DistroInfo {
+    #[serde(default)]
+    name: String,
+    #[serde(default)]
+    version: String,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct CPUInfo {
+    #[serde(default)]
+    vendor: String,
+    #[serde(default)]
+    model: String,
+    #[serde(default)]
+    cpus: usize,
+}
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct MemoryInfo {
+    #[serde(default)]
+    total: u64,
+    #[serde(default)]
+    available: u64,
+    #[serde(default)]
+    free: u64,
+}
+
+pub struct KernelModule {
+    parameters: HashMap<String, String>,
+    desc: String,
+    required: bool,
+}
+pub struct VmContainerCapableDetails {
+    required_cpu_flags: HashMap<String, String>,
+    required_cpu_attribs: HashMap<String, String>,
+    required_kernel_modules: HashMap<String, KernelModule>,
+    cpu_info_file: String,
+}
+
+// Semantic version for the output of the command.
+//
+// XXX: Increment for every change to the output format
+// (meaning any change to the EnvInfo type).
+const FORMAT_VERSION: &str = "1.0.26";
```

Comment:
This is an awkard one. I can see this has been carried over from [the golang implementation](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/cmd/kata-runtime/kata-env.go#L32), but since the format and content of `kata-ctl env` is different to `kata-runtime env`, I think we should probably reset this to `0.0.1` and maybe add a patch suffix (`0.0.1-kata-ctl`) to allow the caller to identify that this is not actually `kata-runtime check` that has suddenly been "downgraded".

See: https://github.com/kata-containers/kata-containers/issues/6090

Reply:
Fixed